fd/jacobian.go

```diff
@@ -0,0 +1,187 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package fd
+
+import (
+	"runtime"
+	"sync"
+
+	"github.com/gonum/floats"
+	"github.com/gonum/matrix/mat64"
+)
+
+type JacobianSettings struct {
+	Formula     Formula
+	OriginValue []float64
+	Step        float64
+	Concurrent  bool
+}
+
+// Jacobian approximates the Jacobian matrix of a vector-valued function f at
+// the location x.
+//
+// The Jacobian matrix J is the matrix of all first-order partial derivatives of f.
+// If f maps an n-dimensional vector x to an m-dimensional vector y = f(x), J is
+// an m×n matrix given as
+//      [ ∂f_1/∂x_1 ... ∂f_1/∂x_n ]
+//      [     .  .          .     ]
+//  J = [     .      .      .     ]
+//      [     .          .  .     ]
+//      [ ∂f_m/∂x_1 ... ∂f_m/∂x_n ]
+// or element-wise
+//  J_ij = ∂f_i/∂x_j.
+//
+// If dst is not nil, the result will be stored in-place into dst and returned,
+// otherwise a new matrix will be allocated first. Finite difference formula and
+// other options are specified by settings. If settings is nil, the Jacobian
+// will be estimated using the Forward formula and a default step size.
+//
+// Jacobian panics if dst is not nil and its size is not m × len(x), or if the
+// derivative order of the formula is not 1.
+func Jacobian(dst *mat64.Dense, f func(y, x []float64), m int, x []float64, settings *JacobianSettings) *mat64.Dense {
```

Comment:
I was looking for something consistent with how Gradient behaves but yes, this would be a good option.